### ISSUE TITLE
[demo] need strict security context for demos

### ISSUE THREAD

**Issue by jmazzitelli** [kiali/kiali#5426]
Try to install travel demo on OpenShift 4.11. You get these warnings:

```
Warning: would violate PodSecurity "restricted:v1.24": allowPrivilegeEscalation != false (container "control" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "control" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "control" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "control" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
```

We need to add securityContext settings to the deployments for the demos - the travel demo for sure and probably [the other ones](https://github.com/kiali/demos), too.

So the pod containers should have the following security contexts:

```yaml
securityContext:
  allowPrivilegeEscalation: false
  capabilities:
    drop:
    - ALL
  privileged: false
  readOnlyRootFilesystem: true
  runAsNonRoot: true
```

**Comment by jshaughn** [comment]
We have a couple of "sub-hack" scripts like create-mesh-namespace.sh and enable_maistra_deployment_injection.sh, maybe we could leverage those to add this new requirement.  They are already used in many of the demo install scripts.

**Comment by jmazzitelli** [comment]
I fixed the travel demo. It also required that i fix that install script which I did here https://github.com/josunect/kiali/pull/2 which is part of this PR https://github.com/kiali/kiali/pull/5397

**Comment by jmazzitelli** [comment]
> maybe we could leverage those to add this new requirement

Maybe this could be utilized: https://github.com/kiali/kiali/blob/master/hack/istio/create-mesh-namespace.sh#L131-L158

But the stuff I changed for this is different - I added `securityContext` to the Deployment container yaml. That is different than what this sub-hack stuff is doing. But I also had to do what this sub-hack is doing - probably should have looked into using it.
